lib/spritely/sprockets/transformer.rb

```diff
@@ -0,0 +1,41 @@
+require 'spritely/version'
+require 'spritely/sprite_map'
+
+module Spritely
+  module Sprockets
+    class Transformer < Struct.new(:input)
+      def self.call(input)
+        new(input).call
+      end
+
+      def self.cache_key
+        @cache_key ||= "#{name}:#{Spritely::VERSION}".freeze
+      end
+
+      def call
+        sprite_map = SpriteMap.new(input[:name].remove("sprites/"), input[:environment], input[:metadata][:sprite_directives])
+
+        data = cache.fetch([self.class.cache_key, input[:name], sprite_map.cache_key]) do
+          sprite_map.files.each do |file|
+            context.depend_on(File.dirname(file))
```

Comment:
Sadly I don't think this is possible, since at preprocessing time we don't yet know about all of the files the sprite map is using.